### PR TITLE
docs: update docs with more guides

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by robszumski** [comment]
These are looking better. It still doesn't say what I do with this. How do I pass it to the machine?

**Comment by crawford** [comment]
I don't think that is relevant to these docs. That is an integration issue and will be covered in coreos/docs.

**Comment by mischief** [comment]
@crawford coreos-cloudinit has lots of examples that start fleetd/etcd2. can you add an ignition example that does that?

File: doc/configuration.md [hunk 1]
```diff
@@ -1,39 +1,7 @@
-# Ignition #
+# Configuration Specification #
```
Comment (crawford) [inline doc/configuration.md]: The spec needs to be cleaned up a bit. It's missing the "passwd" section.

File: doc/providers.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Providers #
```
Comment (crawford) [inline doc/providers.md]: We should drop this entirely. It's only going to confuse our users.

**Comment by crawford** [comment]
Fixes https://github.com/coreos/ignition/issues/115.

File: doc/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
+# Getting Started with Ignition #
+
+Ignition uses a JSON config to represent the set of changes to be made. The
+format of this config is detailed [here][config spec]. One of the most
```
Comment (marineam) [inline doc/getting-started.md]: "in the spec" or similar is better than 90's style "click here" links

File: doc/configuration.md [hunk 4]
```diff
@@ -105,5 +76,35 @@ following section.
...
+      - **system** (boolean): whether or not to make this account a system
+                              account.
+      - **noLogInit** (boolean): whether or not to add the user to the lastlog
+                                 and faillog databases.
+      - **shell** (string): the login shell of the new account.
+  - **groups** (list of objects): the list of groups to be added.
+    - **name** (string): the name of the group.
+    - **gid** (integer): the group ID of the new group.
+    - **passwordHash** (string): the encrypted password of the new group.
+    - **system** (boolean): whether or not to make this group a system group.
```
Comment (marineam) [inline doc/configuration.md]: pretty sure we explicitly removed this option from cloudinit, why is it in ignition?

Comment (crawford) [inline doc/configuration.md]: Removed.

File: doc/getting-started.md (hunk 3 above)
Comment (crawford) [inline doc/getting-started.md]: :space_invader:

**Comment by marineam** [comment]
LGTM

**Pull request merged**